### PR TITLE
cueplayer/sidebar: scroll to active note & remember tab

### PR DESCRIPTION
**Pull request by vojtaholik**
ezgif com-gif-maker (4)

### CODE REVIEW

File: src/pages/video-test.tsx [hunk 1]
```diff
@@ -78,7 +81,21 @@ const EggheadPlayer: React.FC<{videoResource: VideoResource}> = ({
...
+                onClickCue={() => {
+                  playerPrefs.setPlayerPrefs({sideBar: {activeTab: 0}})
+                  setTimeout(() => {
+                    scroller.scrollTo('active-note', {
+                      duration: 0,
+                      delay: 0,
+                      offset: -12,
+                      containerId: 'notes-tab-scroll-container',
+                    })
+                  }, 0.1)
```
Comment (vojtaholik) [inline src/pages/video-test.tsx:94]: Clicking cue:
1. activates `Notes` tab
2. scrolls down to active note.

I couldn't get it working properly in all scenarios until I wrapped scroll function inside very short `setTimeout`. I guess that it's attempting to scroll sooner than `NotesTabContent` exist resulting in no scroll at all. Feels like a hack. 😭

File: src/pages/video-test.tsx [hunk 2]
```diff
@@ -78,7 +81,21 @@ const EggheadPlayer: React.FC<{videoResource: VideoResource}> = ({
...
+                }}
```
Comment (vojtaholik) [inline src/pages/video-test.tsx:95]: Clicking cue:
1. activates `Notes` tab
2. scrolls down to active note.

I couldn't get it working properly in all scenarios until I wrapped scroll function inside very short `setTimeout`. I guess that it's attempting to scroll sooner than `NotesTabContent` exist resulting in no scroll at all. Feels like a hack. 😭

File: src/pages/video-test.tsx [hunk 3]
```diff
@@ -78,7 +81,21 @@ const EggheadPlayer: React.FC<{videoResource: VideoResource}> = ({
                 kind="metadata"
                 label="notes"
               />
-              <CueBar key="cue-bar" order={6.0} />
+              <CueBar
+                onClickCue={() => {
+                  playerPrefs.setPlayerPrefs({sideBar: {activeTab: 0}})
```
Comment (vojtaholik) [inline src/pages/video-test.tsx:86]: I think that making tabs data-driven might make sense now for cases like this. `0` won't always indicate Notes tab.

File: src/pages/video-test.tsx [hunk 4]
```diff
@@ -106,12 +123,21 @@ const EggheadPlayer: React.FC<{videoResource: VideoResource}> = ({
           </div>
           <div className="lg:col-span-3 side-bar">
             <div className="relative h-full">
-              <Tabs className="max-h-[500px] lg:max-h-[none] lg:absolute left-0 top-0 w-full h-full flex flex-col bg-gray-100 dark:bg-gray-1000 text-gray-900 dark:text-white">
+              <Tabs
+                index={playerPrefs.sideBar?.activeTab || 0}
+                onChange={(tabIndex) =>
+                  playerPrefs.setPlayerPrefs({sideBar: {activeTab: tabIndex}})
+                }
```
Comment (vojtaholik) [inline src/pages/video-test.tsx:130]: Get and set active tab using `usePlayerPrefs` based its index as that's what ReachUI Tabs expect.

File: src/pages/video-test.tsx [hunk 5]
```diff
@@ -138,37 +164,39 @@ const NotesTabContent: React.FC<{cues: VTTCue[]}> = ({cues}) => {
...
-            )}
-            {cue.startTime && (
-              <div className="w-full flex items-baseline justify-end pt-3 text-gray-900 dark:text-white">
-                <time className="text-xs opacity-60 font-medium">
-                  {convertTime(cue.startTime)}
-                </time>
-              </div>
-            )}
+          <div key={note.title}>
+            {active && <Element name="active-note" />}
```
Comment (vojtaholik) [inline src/pages/video-test.tsx:172]: This is the element we're scrolling to.

File: src/pages/video-test.tsx (hunk 4 above)
Comment (vojtaholik) [inline src/pages/video-test.tsx:130]: Get and set active tab using `usePlayerPrefs` by its index as that's what ReachUI Tabs expect.

File: src/pages/video-test.tsx (hunk 3 above)
Comment (joelhooks) [reply src/pages/video-test.tsx:86]: give the tabs ids/names instead of generic indices

**Pull request merged**